fitbenchmarking/fitting_benchmarking.py

```diff
@@ -11,82 +11,97 @@
 from fitbenchmarking.utils.logging_setup import logger
 
 from fitbenchmarking.parsing import parse
-from fitbenchmarking.utils import create_dirs, misc
+from fitbenchmarking.utils import create_dirs, misc, options
 from fitbenchmarking.fitbenchmark_one_problem import fitbm_one_prob
 
 
 def fitbenchmark_group(group_name, software_options, data_dir,
                        use_errors=True, results_dir=None):
-    """
-    Gather the user input and list of paths. Call benchmarking on these.
+  """
+  Gather the user input and list of paths. Call benchmarking on these.
+
+  @param group_name :: is the name (label) for a group. E.g. the name for the group of problems in
+                       "NIST/low_difficulty" may be picked to be NIST_low_difficulty
+  @param software_options :: dictionary containing software used in fitting the problem, list of minimizers and
+                             location of json file contain minimizers
+  @param data_dir :: full path of a directory that holds a group of problem definition files
+  @param use_errors :: whether to use errors on the data or not
+  @param results_dir :: directory in which to put the results. None
+                        means results directory is created for you
+
+  @returns :: array of fitting results for the problem group and
+              the path to the results directory
+  """
+
+  logger.info("Loading minimizers from {0}".format(
+      software_options['software']))
+  minimizers, software = misc.get_minimizers(software_options)
+  num_runs = software_options.get('num_runs', None)
+
+  if num_runs is None:
```

Comment:
At the moment, users can define a `software_options` as a  dictionary. If `num_runs` is  not defined in the dictionary then `fitbenchmarking.utils.options` is used to figure out the default option from the json file. This is similar to how the comparison_mode is set:
https://github.com/fitbenchmarking/fitbenchmarking/blob/3685404c13593a4d6c4c116768cf34989cf36a23/fitbenchmarking/results_output.py#L38-L50

I have removed the second default option of 5 for num_runs.

Reply:
Ok, so I think I've figured out why this is confusing me:

line 42 can only be `True` if `software_options['num_runs'] is None`. It seems like this would be a strange situation to occur.  However if it did occur, then line 51 always sets `options_file` to `None` (based on the excerpt above I think your addition should be `options_file = software_options['options_file']`, rather than `sofware_options['num_runs']`).

Reply:
Ah, and if you follow the excerpt above, then your line 50 needs to be `if 'options_file'...` rather than `if 'num_runs'...`

Reply:
Done